internal/controller/customresource/protection_test.go

```diff
@@ -309,7 +310,7 @@ func TestPatchLastConfigAppliedErrors(t *testing.T) {
 			title:        "empty struct cannot be patched",
 			object:       &akov2.AtlasProject{},
 			spec:         &struct{}{},
-			wantErrorMsg: "failed to patch resource:  \"\" is invalid: metadata.name: Required value: name is required",
+			wantErrorMsg: "AtlasProject.atlas.mongodb.com \"\" is invalid: metadata.name: Required value: name is required",
```

Comment:
*nit*: as this uses error contains in the assertion we could make the error more resilient by focusing on the core error msg only, which is that `name is required` or just the bit `metadata.name: Required value: name is required`

Reply:
nit: errors must not start with capital letter

Reply:
this error comes deep from API-machinery, I cannot change it.

It's an architectural smell that we test "Kube logic" in the first place so I am leaving the refactoring to a different contribution.